packages/icons/src/IconNames.ts

```diff
@@ -1675,4 +1678,8 @@ export const enum IconNames {
   SizeLegacy = 'SizeLegacy'
 }
 
+/**
+ * @deprecate IconNames is deprecated.

```

Comment:
Why is this deprecated?

Reply:
Pulling in the type will pull in IconNames, which will result in the issue people are seeing in #7110, so that is why I also marked this deprecated. Does that make sense?

I will talk to Peter about the tool, thanks.

Reply:
More simply put maybe I should have just said it's a deprecated type, since it's keying off a deprecated object. 😄

Reply:
The tool should be adjusted, agreed. We should probably file a bug there as well.

Reply:
@Jahnp - FYI

Reply:
I believe @ThomasMichon added this. A better approach would be to keep the type, but just generate the type. adding this was a cheap way to get some type safety.

also, the tag should be `@deprecated`, not `@deprecate`.

Reply:
I removed this change before the PR was merged.